python/semantic_kernel/connectors/memory/azure_cosmosdb/azure_cosmos_db_memory_store.py

```diff
@@ -0,0 +1,470 @@
+from abc import ABC, abstractmethod
+from typing import List, Tuple
+
+import numpy as np
+from numpy import ndarray
+from pymongo.mongo_client import MongoClient
+
+from semantic_kernel.memory.memory_record import MemoryRecord
+from semantic_kernel.memory.memory_store_base import MemoryStoreBase
+from semantic_kernel.utils.settings import azure_cosmos_db_settings_from_dot_env
+
+# Load environment variables
+(
+    cosmos_api,
+    cosmos_connstr,
+    cosmos_database_name,
+    cosmos_container_name,
+) = azure_cosmos_db_settings_from_dot_env()
+
+# OpenAI Ada Embeddings Dimension
+VECTOR_DIMENSION = 1536
+
+
+# Abstract class similar to the original data store that allows API level abstraction
+class AzureCosmosDBStoreApi(ABC):
+    @abstractmethod
+    async def ensure(self, num_lists, similarity, vector_dimensions):
+        raise NotImplementedError
+
+    @abstractmethod
+    async def create_collection_core(self, collection_name: str) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def get_collections_core(self) -> List[str]:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def delete_collection_core(self, collection_name: str) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def does_collection_exist_core(self, collection_name: str) -> bool:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def upsert_core(self, collection_name: str, record: MemoryRecord) -> str:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def upsert_batch_core(
+        self, collection_name: str, records: List[MemoryRecord]
+    ) -> List[str]:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def get_core(
+        self, collection_name: str, key: str, with_embedding: bool
+    ) -> MemoryRecord:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def get_batch_core(
+        self, collection_name: str, keys: List[str], with_embeddings: bool
+    ) -> List[MemoryRecord]:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def remove_core(self, collection_name: str, key: str) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def remove_batch_core(self, collection_name: str, keys: List[str]) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def get_nearest_matches_core(
+        self,
+        collection_name: str,
+        embedding: ndarray,
+        limit: int,
+        min_relevance_score: float,
+        with_embeddings: bool,
+    ) -> List[Tuple[MemoryRecord, float]]:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def get_nearest_match_core(
+        self,
+        collection_name: str,
+        embedding: ndarray,
+        min_relevance_score: float,
+        with_embedding: bool,
+    ) -> Tuple[MemoryRecord, float]:
+        raise NotImplementedError
+
+
+class MongoStoreApi(AzureCosmosDBStoreApi):
```

Comment:
It would be great if we can place this class to separate file.

Reply:
Moved to a new file.